conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegration.java

```diff
@@ -486,6 +516,9 @@ public void onEvent(SourceEvent.Error event) {
 
     private void handleError(String message) {
         ConvivaSdkConstants.ErrorSeverity severity = ConvivaSdkConstants.ErrorSeverity.FATAL;
+        if (ssai.isAdBreakActive()) {
+            convivaAdAnalytics.reportAdError(message, severity);
+        }
```

Comment:
Do you know if the order is important here?

Also do we need to call `reportAdEnded()` here? It is mentioned on the diagram. Or call it maybe from `ssai.reset()`?

Reply:
I don't think the order is important but I haven't found any documentation regarding this.
Reporting ad ended from `reset` sounds like a sensible thing to do, I'll add it.

Reply:
Based on the workflow diagram `reportAdEnded` is meant to be called on errors.

What I just found is that `adAnalytics.reportAdFailed()` might need to be called as well for non-recoverable errors:
![Screenshot 2024-07-03 at 09 12 44](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/assets/2691371/a817f722-35e0-473a-9d6d-a6872dd8dc01)

Reply:
Unfortunately `reportAdFailed` is not mentioned on the documentation, other than the diagram 😢

Reply:
Looking at the CSAI implementation, `reportAdFailed()` is called on any `AdError`, so we should also call it for SSAI IMO.

Reply:
`reportAdFailed` is reporting a fatal error and ending the ad. I guess we can use either.

Reply:
After digging a bit more it seems that we should be good with only `reportAdError` as we close the session ourselves afterward.

Reply:
[e1d45db](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/73/commits/e1d45db08aa6c5f1d91964dabbdd25088d342601)

Reply:
I guess you meant [2499557](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/73/commits/24995573899f886a8b1c153871663bb8c0bcb2d9)